Mathlib/SetTheory/Ordinal/Exponential.lean

```diff
@@ -21,46 +21,58 @@ universe u v w
 
 namespace Ordinal
 
-/-- The ordinal exponential, defined by transfinite recursion. -/
-instance pow : Pow Ordinal Ordinal :=
-  ⟨fun a b => if a = 0 then 1 - b else limitRecOn b 1 (fun _ IH => IH * a) fun b _ => bsup.{u, u} b⟩
+/-- The ordinal exponential, defined by transfinite recursion.
 
-theorem opow_def (a b : Ordinal) :
-    a ^ b = if a = 0 then 1 - b else limitRecOn b 1 (fun _ IH => IH * a) fun b _ => bsup.{u, u} b :=
-  rfl
+We call this `opow` in theorems in order to disambiguiate from other exponentials. -/
+instance opow : Pow Ordinal Ordinal :=
+  ⟨fun a b ↦ if a = 0 then 1 - b else
+    limitRecOn b 1 (fun _ x ↦ x * a) fun o _ f ↦ ⨆ x : Iio o, f x.1 x.2⟩
+
+private theorem opow_of_ne_zero {a b : Ordinal} (h : a ≠ 0) : a ^ b =
+    limitRecOn b 1 (fun _ x ↦ x * a) fun o _ f ↦ ⨆ x : Iio o, f x.1 x.2 :=
+  if_neg h
 
--- Porting note: `if_pos rfl` → `if_true`
-theorem zero_opow' (a : Ordinal) : 0 ^ a = 1 - a := by simp only [opow_def, if_true]
+private theorem zero_opow' (a : Ordinal) : 0 ^ a = 1 - a :=
```

Comment:
I would keep this one and the previous unprivated (though the porting note can go, as you've done). It's true that the following lemmas characterise the exponential, but it's still useful to have lemmas describing it anyway! For instance, `one_opow` follows from the definitions just as much as `zero_opow'` does, but it's still valuable to have that one.

Reply:
`opow_def` is hideous and I can't imagine it being useful past line 60 of this file.

`zero_opow'` feels to me like an abuse of junk values (specifically the fact `a - b = 0` for `a < b`) but it is used once in this file to golf a result, so I can keep it in if you want.

Reply:
Alright, let's compromise to deprecate `opow_def` (it'll ask for a comment, and you can say that one of `opow_zero`, `opow_succ`, `opow_limit` should be used instead), and keep `zero_opow'`. 

(Now that I think about it, a statement `0 ^ a = if a = 0 then 1 else 0` is sometimes useful... But I'll leave that up to you.)

Reply:
Since the definition of `opow` has changed, I can't easily reprove `opow_def`. In fact, doing so would probably require me to introduce API to link `bsup` with `iSup`... which goes against the entire point of this.